### DISCUSSION TITLE
Set main window size for Windows desktop target

### DISCUSSION THREAD

**Discussion by wgraaf** [unoplatform/uno#11173] [Q&A] [answered]
How can the size of the main window be set when targeting the windows (10) desktop?
The main window starts out way too big, and I have to manually resize every time may app is launched.

**Comment by jeromelaban** [comment]
Thanks for the question. It would be best to ask it in the [WinUI](https://github.com/microsoft/microsoft-ui-xaml) or [WinAppSDK](https://github.com/microsoft/WindowsAppSDK) repositories.

**Comment by Soap-141** [comment, marked as answer]
Make sure you only execute this on Windows.
``` cs
	private void ConfigureViewSize()
	{
		var windowHandle = WinRT.Interop.WindowNative.GetWindowHandle(CurrentWindow);
		var windowId = Microsoft.UI.Win32Interop.GetWindowIdFromWindow(windowHandle);
		var appWindow = Microsoft.UI.Windowing.AppWindow.GetFromWindowId(windowId);
		appWindow.Resize(new Windows.Graphics.SizeInt32 { Width = 480, Height = 800 });
	}
```

**Reply by lucasmontec** [reply]
Where does that Win32Interop comes from? It's not in Uno

**Reply by MartinZikmund** [reply]
As long as you have the `Window` instance, you can just do `window.AppWindow.Resize`

**Comment by wirelessfrank** [comment]
Trying Uno for desktop platform.
Created a new project using blank  template.
I added the suggested code at the end of "OnLaunched" in the App.cs file.
Added 
using Microsoft.UI;
using Microsoft.UI.Windowing;

but I still get: 
Error	CS0234	The type or namespace name 'Win32Interop' does not exist in the namespace 'Microsoft.UI' (are you missing an assembly reference?)

I installed the "Windows App SDK" to get WinUI 3
What Am I doing wrong ?

**Reply by Soap-141** [reply]
Are you using `WinUI` or `UWP`?

**Reply by wirelessfrank** [reply]
I understand I should be using WinUI 3 but there is no choice specific choice in the 
Select startup type.
I tried Blank
Customized to 
1. Framework=> .NET 7.0
2. Platforms => Only Windows
3. Presentation => None
4. Theme => Fluent
5. Extensions => None
6. Projects => None
7. Features => Central Package Management & Visual Studio Code Debugging
8. Authentication => None
9. Application =>com.companyname.UnoApp1
10. Testing => None

I go to the Apps.cs file and get the result above.

**Reply by Soap-141** [reply]
Try this @wirelessfrank 
![image](https://github.com/unoplatform/uno/assets/57812285/1fe18933-7d3f-4556-bd32-86705c16e6e3)

Else here is my sample. (Forgot to rename the window name but you should be good)
[WindowSize.zip](https://github.com/unoplatform/uno/files/12525751/WindowSize.zip)
